### PR TITLE
Turborepo + Drizzle + Auth Proxy + Docs

### PR DESCRIPTION
**Pull request by RishikeshNK**
## Description

I spent way too much time on this _small_ PR.

TLDR:
- Migrate from Prisma to Drizzle
- Makes the project a monorepo (Turborepo + PNPM Workspace)
- Adds an Auth Proxy using Nitro to support OAuth in Vercel Preview environments
- Adds a basic documentation site to onboard new devs!

## Links
[Cooper Turbo](https://github.com/RishikeshNK/cooper-turbo)
[Cooper Turbo Deployed](https://cooper-turbo.vercel.app/)
[Cooper Docs](https://cooper-turbo-docs.vercel.app/)

**Note:** Deployment is going to be broken so ignore the build. The build is passing [here](https://github.com/RishikeshNK/cooper-turbo/actions/runs/10745516626). I'll have to spend some time setting up the _three_ new projects on Sandbox Vercel - main app, auth proxy, and the docs site but it shouldn't be a lot of work since I did it on the repo on my account.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (joshw2048) [inline .github/pull_request_template.md:1]: Once we have tests, we should add a testing section to the pr template

File: README.md [hunk 1]
```diff
@@ -4,25 +4,34 @@ Cooper is a tool for Northeastern students to both submit reviews of their co-op
...
 - [shadcn/ui](https://ui.shadcn.com/) - Pre-built components
 - [NextAuth.js](https://next-auth.js.org/) - Authentication
 - [Prisma](https://www.prisma.io/) - ORM
 - [PostgreSQL](https://www.postgresql.org/) - Relational database
-- [Docker](https://www.docker.com/) - Containerization
-- [tRPC](https://trpc.io/) - End-to-end typesafe API
+- [Docker](https://www.docker.com/) - Database containerization for local development
 - [Zod](https://zod.dev/) - Validation
+- [Jest](https://jestjs.io/) - Unit tests
+- [Playwright](https://playwright.dev/) - E2E tests
```
Comment (joshw2048) [inline README.md:17]: i assume playwright/jest will be part of our github actions pipeline once they're built? for the time being, can we indicate that these aren't set up yet in the readme?

File: apps/docs/docs/03-backend/database.mdx
Comment (joshw2048) [inline apps/docs/docs/03-backend/database.mdx:1]: these docs are fire, did you write these yourself? or is this auto generated with some sort of tool

File: apps/web/README.md
Comment (joshw2048) [inline apps/web/README.md:1]: do we need this file

File: apps/web/src/app/(pages)/(protected)/review/page.tsx
Comment (joshw2048) [inline apps/web/src/app/(pages)/(protected)/review/page.tsx:1]: soooo can you gave me the explain like im 5 of how turborepo/pnpm workspace turned the project into a monorepo? is a lot of the changes in this PR caused by turbo repo or did you do it manually?

**Review by joshw2048 — APPROVED** [review]
i gave up looking through all the files after 50 of them, but tbh lgtm. approved to merge for now, let's sit down sometime and talk about these changes just so i understand

+ would be an interesting topic for some tech design review or somethjing

File: .github/pull_request_template.md
Comment (RishikeshNK) [reply .github/pull_request_template.md:1]: Yup good point! We can add that to the template once we start adding unit and E2E tests.

**Review by banushi-a — COMMENTED** [review]
So I did look through most of these files but things look pretty good. We can talk more in depth during kickoff tomorrow. lgtsm

File: apps/docs/docs/03-backend/database.mdx
Comment (RishikeshNK) [reply apps/docs/docs/03-backend/database.mdx:1]: The docs are built on top of Docusaurus but I wrote all of the content 🤓

File: apps/web/src/app/(pages)/(protected)/review/page.tsx
Comment (RishikeshNK) [reply apps/web/src/app/(pages)/(protected)/review/page.tsx:1]: In short, all the files within the previous `app` folder was moved to `packages` and `apps`. We have separate packages for the API, database, UI, and authentication. We also have three apps - the Cooper web app, authentication proxy for OAuth, and the docs. It's a monorepo now because there are multiple "projects" in this single repo. A lot of the additions are because of boilerplate code from [create-t3-turbo](https://github.com/t3-oss/create-t3-turbo).

File: apps/web/README.md
Comment (RishikeshNK) [reply apps/web/README.md:1]: I was thinking of having these docs for specifically the web app -- we can modify it in the future.

**Comment by RishikeshNK** [comment]
pray 4 me 🙏🏻
